project_generator/tools/gccarm.py

```diff
@@ -193,8 +193,12 @@ def build_project(self):
         else:
             if ret_code != self.SUCCESSVALUE:
                 # Seems like something went wrong.
-                logging.error("Build failed with the status: %s" %
-                              self.ERRORLEVEL[ret_code])
+                if ret_code < 3:
+                    logging.error("Build failed with the status: %s" %
+                                  self.ERRORLEVEL[ret_code])
+                else:
+                    logging.error("Build failed with unknown error. Returned: %s" %
```

Comment:
Note : this one is a problem in make in tests, it raises an exception ,I'll look at this and this else should be removed ;)